src/Http/Middleware/AddIdpUserDataMiddleware.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Zanichelli\IdpExtension\Http\Middleware;
+
+use Closure;
+use Exception;
+use GuzzleHttp\Client;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Zanichelli\IdpExtension\Models\ZTrait\ZUserBuilder;
+
+class AddIdpUserDataMiddleware
+{
+    use ZUserBuilder;
+
+    public function handle(Request $request, Closure $next, string $idpUrlEnvKey = 'IDP_URL')
+    {
+        $token = $request->cookies->get(config("idp.cookie.name"));
+        if ($token) {
+            try {
+                $client = new Client(['verify' => false]);
+                $res = $client->get(env($idpUrlEnvKey) . '/v1/user?token=' . $token);
+                $request->merge(['user' => json_decode($res->getBody(), true)]);
```

Comment:
In questo modo si ottiene un array con alcune chiavi differenti rispetto a quelle dell'altro middleware (per esempio `is_verified` invece di `isVerified`), forse sarebbe comodo avere una struttura comune?

Reply:
Però mi pare che le informazioni restituite nei due casi non siano esattamente sovrapponibili, cioè ci sono dati diversi, non tutti sono presenti in entrambi i casi.

Reply:
Mi pare che le uniche cose che non sono presenti in entrambi i casi siano i permessi (che però sono comunque opzionali), il token e l'iss. La classe `ZUser` non ha neppure il campo `is_myz`, però forse avrebbe senso che ci fosse.